controller/replica_controller.go

```diff
@@ -420,6 +408,80 @@ func (rc *ReplicaController) CreateInstance(obj interface{}) (*types.InstancePro
 	return c.ReplicaProcessCreate(r.Name, r.Spec.EngineImage, dataPath, backingImagePath, r.Spec.VolumeSize, r.Spec.RevisionCounterDisabled)
 }
 
+func (rc *ReplicaController) GetBackingImagePathForReplicaStarting(r *longhorn.Replica) (string, error) {
+	log := getLoggerForReplica(rc.logger, r)
+
+	bi, err := rc.ds.GetBackingImage(r.Spec.BackingImage)
+	if err != nil {
+		return "", err
+	}
+	if bi.Status.UUID == "" {
+		log.Debugf("The requested backing image %v has not been initialized, UUID is empty", bi.Name)
+		return "", nil
+	}
+	if bi.Spec.Disks == nil {
+		log.Debugf("The requested backing image %v has not started disk file handling", bi.Name)
+		return "", nil
+	}
+	if _, exists := bi.Spec.Disks[r.Spec.DiskID]; !exists {
+		bi.Spec.Disks[r.Spec.DiskID] = struct{}{}
+		log.Debugf("Replica %v will ask backing image %v to download file to node %v disk %v",
+			r.Name, bi.Name, r.Spec.NodeID, r.Spec.DiskID)
+		if _, err := rc.ds.UpdateBackingImage(bi); err != nil {
+			return "", err
+		}
+		return "", nil
+	}
+	// bi.Spec.Disks[r.Spec.DiskID] exists
+	if fileStatus, exists := bi.Status.DiskFileStatusMap[r.Spec.DiskID]; !exists || fileStatus.State != types.BackingImageStateReady {
+		currentBackingImageState := ""
+		if fileStatus != nil {
+			currentBackingImageState = string(fileStatus.State)
+		}
+		log.Debugf("Replica %v is waiting for backing image %v downloading file to node %v disk %v, the current state is %v",
+			r.Name, bi.Name, r.Spec.NodeID, r.Spec.DiskID, currentBackingImageState)
+		return "", nil
+	}
+	return types.GetBackingImagePathForReplicaManagerContainer(r.Spec.DiskPath, r.Spec.BackingImage, bi.Status.UUID), nil
+}
+
+func (rc *ReplicaController) CanStartRebuildingReplica(r *longhorn.Replica) (bool, error) {
+	log := getLoggerForReplica(rc.logger, r)
+
+	concurrentRebuildingLimit, err := rc.ds.GetSettingAsInt(types.SettingNameConcurrentReplicaRebuildPerNodeLimit)
```

Comment:
What's the default setting for `concurrentRebuildingLimit`? I assume it will be `0`.

If it's the default setting, probably we just skip the following check and return `true` to allow the rebuild process.

Reply:
OK. I will update the default value to 0 to avoid regression.